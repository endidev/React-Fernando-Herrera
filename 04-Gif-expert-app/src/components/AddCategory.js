import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue( e.target.value )
    }

    const handdleSubmit = (e) => {
        e.preventDefault();
        
        if(inputValue.trim().length > 2){
            setCategories( cats => [inputValue, ...cats])
        }

    }

    return (
        <form onSubmit={handdleSubmit}>
        
            <input 
                type='text'
                value= {inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
