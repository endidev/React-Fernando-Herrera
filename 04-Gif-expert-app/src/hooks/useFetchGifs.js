import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs"
import PropTypes from 'prop-types';


export const useFetchGifs = ( category ) =>  {
    const [state, setState] = useState({
        data: [],
        loading: true,
    })

    useEffect( () => {
        getGifs( category )
            .then( imgs => {
                setState({
                    data: imgs,
                    loading: false
                })
            })
    }, [category])

    return state; //data [], loading:true;
}


useFetchGifs.propTypes = {
    category: PropTypes.string.isRequired
}