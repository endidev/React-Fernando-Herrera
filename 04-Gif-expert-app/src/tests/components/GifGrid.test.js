import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('hook fetch', () => {
    
    const category = 'One Punch';
    

    test('Compararlo con snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={category}/>)
        expect( wrapper).toMatchSnapshot();
    })

    test('debe mostrar items al cargar  imagenes con useFetchGifs', () => {
        
        const gifs = [{
            id: 'abc',
            url: 'https://localhost/cualquier/cosa.jgp',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={category}/>)

        expect( wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    })
    
    
})