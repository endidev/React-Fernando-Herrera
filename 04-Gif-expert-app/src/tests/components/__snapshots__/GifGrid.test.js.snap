// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hook fetch Compararlo con snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`hook fetch debe mostrar items al cargar  imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jgp"
    />
  </div>
</Fragment>
`;
